src/masternode.cpp

```diff
@@ -189,14 +193,26 @@ void CMasternode::Check(bool forceCheck)
             TRY_LOCK(cs_main, lockMain);
             if(!lockMain) return;
 
-            if(!AcceptToMemoryPool(mempool, state, CTransaction(tx), false, NULL, false, true, true)){
-                activeState = MASTERNODE_VIN_SPENT;
+            if(!AcceptToMemoryPool(mempool, state, CTransaction(tx), false, NULL, false, true, true)) {
+                nActiveState = MASTERNODE_OUTPOINT_SPENT;
                 return;
             }
         }
     }
 
-    activeState = MASTERNODE_ENABLED; // OK
+    nActiveState = MASTERNODE_ENABLED; // OK
+}
+
+std::string CMasternode::GetStatus()
+{
+    switch(nActiveState) {
+        case CMasternode::MASTERNODE_PRE_ENABLED:       return "PRE_ENABLED";
+        case CMasternode::MASTERNODE_ENABLED:           return "ENABLED";
+        case CMasternode::MASTERNODE_EXPIRED:           return "EXPIRED";
+        case CMasternode::MASTERNODE_OUTPOINT_SPENT:    return "OUTPOINT_SPENT";
```

Comment:
Isn't the vin terminology a pretty strongly established usage though, including among users ?

Also I'm not sure calling it vin is that wrong because there's a 1:1 relationship between a CTxOut and a CTxIn.  The CTxIn just adds some additional information.